### PR TITLE
reformat whole SC docstrings draft

### LINKED ISSUE #371: `field/` module is full of files with misformatted docstrings
# Problem
Most (if not all) files that have been created do not adhere to the style guides as linked in the README (https://realpython.com/documenting-python-code/ and https://numpydoc.readthedocs.io/en/latest/format.html). This is a pretty significant problem that could really bog the process of merging `SC_redesign` into `master` when the soil and crop redesign has been completed.
# Solution
This issue should be addressed with a series of PR's that reformat the comments of groups of files at a time. The size of these sets and the best way to group them should be discussed. I think reformatting 1 - 2 files at a time would make this process drag on for too long, and reviewing entire sub-modules at a time (`soil`, `crop`, `field`, `tests_SC`) would result in PRs that are too large and broad.
# Strategy
- Start executing these PRs as soon as possible, in tandem with starting new refactors of old code. The reason for this is that someone who has just corrected the comments in an older file will know exactly what the comments and docstrings should look like and will be much less likely to make the same mistakes while working on a new file.
- Regularly have people from outside the Soil and Crop team participate in the review process, so that we know the reformatted files are meeting the style standards of the entire RuFaS team.

### PR DESCRIPTION
**Pull request by matthew7838**
### Summary
Docstrings developed in the early stage of reactors were in Google format, this PR simply converts them into the requested Numpy format.

### Notes
Please do take a look into it, they're mostly generated by chatGPT, but with the amount of old docstrings, I might not be able to catch all the errors between the conversions.

### CODE REVIEW

File: RUFAS/routines/field/soil/infiltration.py [hunk 1]
```diff
@@ -116,44 +116,65 @@ def infiltrate(self, rainfall: float, weighting_coefficient: float, potential_ev
...
-            curve number 1 (dry/wilting point conditions) (unitless)
+        Returns
+        -------
+        float
+            Curve number 1 (dry/wilting point conditions) (unitless).
 
-        SWAT Reference: 2:1.1.4
+        SWAT Reference
+        --------------
+        2:1.1.4
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]: References should be written as
```python
References
----------
SWAT Theoretical documentation eqn. 2:1.1.4
```
to keep them consistent with the references sections of other docstrings in the Crop and Soil module.

File: RUFAS/routines/field/soil/soil_config_factory.py [hunk 2]
```diff
@@ -18,8 +18,20 @@ class SoilConfigFactory:
     @staticmethod
     def create_soil_data(field_size: float, config: SoilConfiguration = SoilConfiguration("generic"),
                          **kwargs) -> SoilData:
-        """Creates a soil data object from a SoilConfiguration enum, with the defaults from that configurations and the
-            optional ability to modify attributes
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_config_factory.py]: Top level description is left out here.

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 3]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
-        Returns:
-            coarseness factor of erodibility, based on sand content (unitless)
+        Returns
+        -------
+        float
+            Coarseness factor of erodibility, based on sand content (unitless).
 
-        SWAT Reference: 4:1.1.6
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 4]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
-        Returns:
-            The clay-silt ratio factor, based on clay and silt content (unitless)
+        Returns
+        -------
+        float
+            The clay-silt ratio factor, based on clay and silt content (unitless).
 
-        SWAT Reference: 4:1.1.7
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 5]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
-        Returns:
-            Carbon factor of erodibility based on organic carbon content of the soil (unitless)
+        Returns
+        -------
+        float
+            The carbon factor of erodibility based on the organic carbon content of the soil (unitless).
 
-        SWAT Reference: 4:1.1.8
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 6]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
-        Returns:
-            The high sand content factor of erodibility, based on the amount of sand in the soil (unitless)
+        Returns
+        -------
+        float
+            The high sand content factor of erodibility, based on the amount of sand in the soil (unitless).
 
-        SWAT Reference: 4:1.1.9
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 7]
```diff
@@ -153,24 +193,7 @@ def _determine_high_sand_factor(percent_sand_content: float) -> float:
...
-        Details:
-            Some soils are more prone to erosion than others, based on how much sand, silt, clay, and organic carbon
-            they contain.
-
-        Returns:
-            The soil erodibility factor based on its coarse sand content, clay-silt ratio, clay content, and organic
-            carbon content (compound unit, irrelevant)
 
-        SWAT Reference: 4:1.1.5
-        """
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]: Missing docstring here

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 8]
```diff
@@ -179,21 +202,30 @@ def _determine_soil_erodibility_factor(percent_sand_content: float, percent_silt
...
-        Returns:
-            The cover and management factor (unitless)
+        Returns
+        -------
+        float
+            The cover and management factor (unitless).
 
-        SWAT Reference: 4:1.1.10
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 9]
```diff
@@ -210,38 +242,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
     @staticmethod
     def _determine_exponential_term(average_subbasin_slope: float) -> float:
-        """Calculates the exponential term, which is used to calculate the topographic factor
+        """
+        Calculate the exponential term used to determine the topographic factor.
 
-        Args:
-            average_subbasin_slope: average slope fraction of the subbasin (unitless)
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 10]
```diff
@@ -210,38 +242,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
+        average_subbasin_slope : float
+            Average slope fraction of the subbasin (unitless).
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Returns
        -------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 11]
```diff
@@ -210,38 +242,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
+        slope to that from a 22.1 m length of uniform 9% slope under otherwise identical conditions.
 
-        Details:
-            The length and slope of a soil have an effect on erodibility, with shorter lengths and steeper slopes
-            resulting in more soil erosion. A shorter length means more erosion because sediment does not have to travel
-            as far to reach channels and be removed from the soil. A steeper slope means that there is less
-            resistance against the gravitational forces acting on a piece of sediment as it moves down the slope,
-            resulting in the soil eroding more easily.
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 12]
```diff
@@ -210,38 +242,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
+        slope_length : float
+            Length of the slope (m).
+        average_subbasin_slope : float
+            Average slope fraction of the subbasin (m rise over m run).
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Returns
        -------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 13]
```diff
@@ -210,38 +242,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
-            The topographic factor (unitless)
+        --------
+        float
+            The topographic factor (unitless).
 
-        SWAT Reference: 4:1.1.12
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 14]
```diff
@@ -253,20 +293,27 @@ def _determine_topographic_factor(slope_length: float, average_subbasin_slope: f
...
     @staticmethod
     def _determine_coarse_fragment_factor(percent_rock_content: float) -> float:
-        """Calculates coarse fragment factor for use in calculating sediment yield
+        """
+        Calculate the coarse fragment factor for use in calculating sediment yield.
 
-        Args:
-            percent_rock_content: percent rock in first soil layer
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 15]
```diff
@@ -253,20 +293,27 @@ def _determine_topographic_factor(slope_length: float, average_subbasin_slope: f
...
+        percent_rock_content : float
+            Percent rock in the first soil layer.
 
         Details:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Details
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 16]
```diff
@@ -253,20 +293,27 @@ def _determine_topographic_factor(slope_length: float, average_subbasin_slope: f
...
-            Since the mass of rocks are so much greater than particles of sand, silt, clay, etc., they require
-            significantly more force to move them off of a field, in other words erode them. So by having more rocks in
-            a soil, that soil erodes slower.
+        --------
+        The mass of rocks is much greater than particles of sand, silt, clay, etc., requiring significantly more force
+        to move them off a field, resulting in slower erosion in soils with higher rock content.
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Returns
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 17]
```diff
@@ -253,20 +293,27 @@ def _determine_topographic_factor(slope_length: float, average_subbasin_slope: f
...
-            Coarse fragment factor based on the rock content of the soil (unitless)
+        --------
+        float
+            Coarse fragment factor based on the rock content of the soil (unitless).
 
-        SWAT Reference: 4:1.1.15
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 18]
```diff
@@ -474,39 +521,58 @@ def _determine_sediment_yield(surface_area_runoff: float, peak_runoff_rate: floa
...
-            support_practice_factor: ratio of soil loss with specific support practice to corresponding loss with
-                up-and-down slope culture (unitless)
-            topographic_factor: expected ratio of soil loss per unit area from a field slope to that from a 22.1 m
-                length of uniform 9% slope under identical conditions (unitless)
-            coarse_fragment_factor: factor that adjusts for percent rock in the first soil layer (unitless)
+        """
+        Calculate the sediment yield for a given day.
+
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 19]
```diff
@@ -474,39 +521,58 @@ def _determine_sediment_yield(surface_area_runoff: float, peak_runoff_rate: floa
...
     def _determine_adjusted_sediment_yield(sediment_yield: float, snow_water_content: float) -> float:
-        """Adjusts the sediment yield based on how much snow cover there is
+        """
+        Adjust the sediment yield based on the amount of snow cover.
 
-        Args:
-            sediment_yield: sediment yield on a given day (metric tons)
-            snow_water_content: water content of the snow cover (mm)
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 20]
```diff
@@ -474,39 +521,58 @@ def _determine_sediment_yield(surface_area_runoff: float, peak_runoff_rate: floa
...
+        sediment_yield : float
+            Sediment yield on a given day (metric tons).
+        snow_water_content : float
+            Water content of the snow cover (mm).
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Returns
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 21]
```diff
@@ -474,39 +521,58 @@ def _determine_sediment_yield(surface_area_runoff: float, peak_runoff_rate: floa
...
-            The sediment yield on a given day adjusted for water content of the snow cover
+        --------
+        float
+            The sediment yield on a given day adjusted for the water content of the snow cover.
 
-        SWAT Reference: 4:1.3.1
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 22]
```diff
@@ -27,31 +27,43 @@ def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] =
...
-            min_temp: minimum temperature of the current day (degrees C)
-            max_temp: maximum temperature of the current day (degrees C)
-            plant_cover: total aboveground plant biomass and residue on the current day (kg per hectare)
-            snow_cover: water content of the snow cover on the current day (mm)
-            avg_annual_air_temp: average annual air temperature (degrees C)
+        """
+        Update the soil temperature.
+
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 23]
```diff
@@ -27,31 +27,43 @@ def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] =
...
+        that can occur throughout the year.
+
+        Because this model now simulates the top 20 mm of the soil profile as its own layer for the purpose of
+        tracking phosphorus runoff, this module first performs operations specifically on the top 2 layers of soil.
+        The result is that this module treats the top two layers of soil as if they were 1, i.e. both layers are set
+        to be the same temperature every day. For every day of the simulation besides potentially the first, the
+        top two layers of soil will have the same temperature.
+
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 24]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
     @staticmethod
     def _determine_maximum_damping_depth(bulk_density: float) -> float:
-        """calculates maximum damping depth of a soil profile based on bulk density
+        """
+        Calculate the maximum damping depth of a soil profile based on bulk density.
 
-        Args:
-            bulk_density: the soil profile bulk density (Mg per cubic meter)
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 25]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
+        bulk_density : float
+            The soil profile bulk density (Mg per cubic meter).
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Returns
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 26]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
-            the maximum damping depth (mm)
+        --------
+        float
+            The maximum damping depth (mm).
 
-        SWAT Reference: 1:1.3.6
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 27]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
-        """calculates the scaling factor for use in calculating the damping depth
+        """
+        Calculate the scaling factor for use in calculating the damping depth.
 
-        Args:
-            soil_water_content: amount of water in soil profile expressed as depth of water in profile (mm)
-            bulk_density: bulk density of the soil profile (Mg per cubic meter)
-            bottom_depth: depth from the soil surface of the bottom of the soil profile (mm)
+        Parameters:
+        -----------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Parameters
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 28]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
+        soil_water_content : float
+            Amount of water in the soil profile expressed as depth of water in profile (mm).
+        bulk_density : float
+            Bulk density of the soil profile (Mg per cubic meter).
+        bottom_depth : float
+            Depth from the soil surface of the bottom of the soil profile (mm).
 
         Returns:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Returns
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 29]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
-            the scaling factor for calculating damping depth (unitless)
+        --------
+        float
+            The scaling factor for calculating damping depth (unitless).
 
-        SWAT Reference: 1:1.3.7
+        Reference:
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 30]
```diff
@@ -98,65 +110,97 @@ def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float,
...
+        SWAT 1:1.3.7
         """
         return soil_water_content / ((0.356 - (0.144 * bulk_density)) * bottom_depth)
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py:158]: Rest of docstrings in this file need to be corrected following above suggested conventions

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]
These docstring look good Matthew. There are a few files where titles of docstring subsections need to be rewritten, and the SWAT references should be a little more specific overall.

File: RUFAS/routines/field/soil/soil_temp.py (hunk 30 above)
Comment (matthew7838) [reply RUFAS/routines/field/soil/soil_temp.py:158]: Although this is still a draft, thanks for the comments, Ed. I'll be fixing these along with sphinx issues.

File: RUFAS/routines/field/soil/infiltration.py [hunk 31]
```diff
@@ -116,45 +116,66 @@ def infiltrate(self, rainfall: float, weighting_coefficient: float, potential_ev
...
-        Returns:
-            curve number 3 (wet/field capacity conditions) (unitless)
+        Returns
+        -------
+        float
+            Curve number 3 (wet/field capacity conditions) (unitless).
 
-        SWAT Reference: 2:1.1.5
+        Reference
+        --------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/infiltration.py [hunk 32]
```diff
@@ -215,77 +252,113 @@ def _determine_retention_parameter(soil_water_content: float,
...
+        Returns
+        -------
+        float
+            Accumulated runoff or rainfall excess (mm).
 
-        Details:
-            Runoff only occurs when rainfall is greater than initial abstractions (about surface storage, interception,
-            etc.) which are approximated as 0.2 * retention parameter in SWAT 2:1.1 and here
+        Notes
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]:
```suggestion
        Notes
        -----
```

File: RUFAS/routines/field/soil/infiltration.py [hunk 33]
```diff
@@ -215,77 +252,113 @@ def _determine_retention_parameter(soil_water_content: float,
...
+        Runoff only occurs when rainfall is greater than initial abstractions (about surface storage, interception, etc.)
+        which are approximated as 0.2 times the retention parameter in SWAT 2:1.1 and here.
 
-        SWAT Reference: 2:1.1.1, 3
+        References
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/infiltration.py [hunk 34]
```diff
@@ -299,22 +372,33 @@ def _determine_updated_retention_parameter(previous_retention_parameter: float,
...
+            Weighting coefficient used to calculate the retention coefficient for daily curve number calculations
+            dependent on plant evapotranspiration (unitless).
+
+        Returns
+        -------
+        float
+            Retention parameter for the current day (mm).
+
+        References
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/infiltration.py [hunk 35]
```diff
@@ -324,14 +408,21 @@ def _determine_updated_retention_parameter(previous_retention_parameter: float,
...
-        Returns:
-            the curve number for a given day adjusted for moisture content (unitless)
+        Returns
+        -------
+        float
+            The curve number for a given day adjusted for moisture content (unitless).
 
-        SWAT Reference: 2:1.1.11
+        References
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/infiltration.py]:
```suggestion
        References
        ----------
```

File: RUFAS/routines/field/soil/soil_data.py [hunk 36]
```diff
@@ -363,18 +376,33 @@ def profile_saturation(self) -> float:
...
-        """Returns: the soil water factor (unitless)
-        SWAT Reference: 5:2.3.18
+        """
+        Get the soil water factor (unitless).
+
+        Returns
+        -------
+        float
+            The soil water factor (unitless)
+        References
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_data.py]:
```suggestion
            The soil water factor (unitless)
            
        References
```

File: RUFAS/routines/field/soil/soil_data.py [hunk 37]
```diff
@@ -363,18 +376,33 @@ def profile_saturation(self) -> float:
...
+        def get_field_capacity_soil_water_content():
+            """
+            Calculate the total amount of water contained in the entire soil profile at field capacity
+            (but not saturated) (mm).
+
+            Returns
+            -------
+            float
+                The soil water content at field capacity in millimeters (mm).
+            """
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_data.py]: Indentation is off here, also the method name should not be changed.

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 38]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
+        percent_silt_content : float
+            Percent of silt in the given layer of soil.
+        percent_clay_content : float
+            Percent of clay in the given layer of soil.
 
-        Details:
-            The clay-silt ratio effects the erodibility of the soil, specifically soils with a high ratio of clay to
-            silt are less susceptible to erosion than soils with lower ratios of clay to silt.
+        Notes
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Notes
        -----
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 39]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
+        Parameters
+        ----------
+        percent_organic_carbon : float
+            The percent of organic carbon content in the given layer of soil.
 
-        Details:
-            The amount of organic carbon content in the soil effects the erodibility of the soil, specifically soils
-            with higher amounts of organic carbon content will have less erosion than soils with less.
+        Notes
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py:150]:
```suggestion
        Notes
        -----
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 40]
```diff
@@ -74,77 +79,112 @@ def erode(self, field_size: float, minimum_cover_management_factor: float, surfa
...
-            percent_sand_content: percent of sand in the given layer of soil
+        Parameters
+        ----------
+        percent_sand_content : float
+            The percent of sand in the given layer of soil.
 
-        Details:
-            When a soil has an extremely high sand content, it reduces the erodibility of that soil.
+        Notes
+        -------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Notes
        -----
```

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 41]
```diff
@@ -210,38 +271,46 @@ def _determine_support_practice_factor() -> float:  # TODO: implement this for v
...
+        """
+        Calculate the topographic factor, which represents the expected ratio of soil loss per unit area from a field
+        slope to that from a 22.1 m length of uniform 9% slope under otherwise identical conditions.
 
-        Details:
-            The length and slope of a soil have an effect on erodibility, with shorter lengths and steeper slopes
-            resulting in more soil erosion. A shorter length means more erosion because sediment does not have to travel
-            as far to reach channels and be removed from the soil. A steeper slope means that there is less
-            resistance against the gravitational forces acting on a piece of sediment as it moves down the slope,
-            resulting in the soil eroding more easily.
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py:239]: This should be included as Notes

File: RUFAS/routines/field/soil/soil_erosion.py [hunk 42]
```diff
@@ -253,20 +322,27 @@ def _determine_topographic_factor(slope_length: float, average_subbasin_slope: f
...
-            a soil, that soil erodes slower.
+        Notes
+        -----
+        The mass of rocks is much greater than particles of sand, silt, clay, etc., requiring significantly more force
+        to move them off a field, resulting in slower erosion in soils with higher rock content.
 
-        Returns:
-            Coarse fragment factor based on the rock content of the soil (unitless)
+        Returns
+        --------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        Returns
        -------
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 43]
```diff
@@ -27,31 +27,43 @@ def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] =
...
+        max_temp : float
+            Maximum temperature of the current day (degrees C).
+        plant_cover : float
+            Total aboveground plant biomass and residue on the current day (kg per hectare).
+        snow_cover : float
+            Water content of the snow cover on the current day (mm).
+        avg_annual_air_temp : float
+            Average annual air temperature (degrees C).
 
         Notes:
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        Notes
```

File: RUFAS/routines/field/soil/soil_temp.py [hunk 44]
```diff
@@ -165,49 +209,75 @@ def _determine_depth_factor(center_depth: float, damping_depth: float) -> float:
...
-        Returns:
-            bare soil surface temperature (degrees C)
+        Returns
+        -------
+        float
+            Bare soil surface temperature (degrees C).
 
-        SWAT Reference: 1:1.3.9
+        References
+        ---------------
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_temp.py]:
```suggestion
        References
        ----------
```

**Review by Isgotkowitz — APPROVED** [review]
Looks good, left comments on some typos I spotted, once those get corrected should be good to go.

**Review by JoeWaddell — APPROVED** [review]
With corrections following the suggestions/comments by @Isgotkowitz , LGTM

**Pull request merged**